manifests/tools.pp

```diff
@@ -0,0 +1,206 @@
+    match => '^db=.*|^db\s.*',
+  }
+  file_line { 'backup_and_sync/config db_password':
+    path  => '/root/grafana_tools/backup_and_sync/config',
+    line  => "db_password=${::profile_metrics::db_passwd}",
+    match => '^db_password.*',
+  }
+  file_line { 'backup_and_sync/config db_user':
+    path  => '/root/grafana_tools/backup_and_sync/config',
+    line  => "db_user=${::profile_metrics::db_user}",
+    match => '^db_user.*',
+  }
+
+  # bulk_alert/config
+  file_line { 'bulk_alert/config user':
+    path  => '/root/grafana_tools/bulk_alert/config',
+    line  => "user=${grafana_admin_user}",
+    match => '^user.*',
+  }
+  file_line { 'bulk_alert/config pass':
+    path  => '/root/grafana_tools/bulk_alert/config',
+    line  => "pass=${grafana_admin_password}",
+    match => '^pass.*',
+  }
+
+  # ldap_sync/config
+  file_line { 'ldap_sync/config admin_password':
+    path  => '/root/grafana_tools/ldap_sync/config',
+    line  => "admin_password=\"${grafana_admin_password}\"",
+    match => '^admin_password.*',
+  }
+  file_line { 'ldap_sync/config base_dir':
+    path  => '/root/grafana_tools/ldap_sync/config',
+    line  => "base_dir=\"${ldap_sync_base_dir}\"",
+    match => '^base_dir.*',
+  }
+  $ldap_groups_joined = $ldap_sync_groups.unique.sort.join(' ')
+  file_line { 'ldap_sync/config groups':
+    path  => '/root/grafana_tools/ldap_sync/config',
+    line  => "groups=(${ldap_groups_joined})",
+    match => '^groups.*',
+  }
+
+  # white_label/config
+  if ( ! empty($whitelabel_title) ) {
+    file_line { 'white_label/config title':
+      path  => '/root/grafana_tools/white_label/config',
+      line  => "title=\"${whitelabel_title}\"",
+      match => '^title.*',
+    }
+  }
+  if ( ! empty($whitelabel_subtitle) ) {
+    file_line { 'white_label/config subtitle':
+      path  => '/root/grafana_tools/white_label/config',
+      line  => "subtitle=\"${whitelabel_subtitle}\"",
+      match => '^subtitle.*',
+    }
+  }
+
+  # ADD CRONS FOR SYNC SCRIPTS
+  if ( $backup_and_sync_enable ) {
+    cron { 'Grafana_tools backup_and_sync sync_grafana':
+      command  => '( cd /root/grafana_tools/backup_and_sync/ && ./sync_grafana.sh ) 2>&1',
+      user     => 'root',
+      hour     => $backup_and_sync_cron_hour,
+      minute   => $backup_and_sync_cron_minute,
+      weekday  => $backup_and_sync_cron_weekday,
+      month    => $backup_and_sync_cron_month,
+      monthday => $backup_and_sync_cron_monthday,
+    }
+  }
+  else {
+    cron { 'Grafana_tools backup_and_sync sync_grafana':
+      ensure => absent,
+    }
+  }
+  if ( $ldap_sync_enable ) {
+    cron { 'Grafana_tools ldap_sync':
+      command  => '( cd /root/grafana_tools/ldap_sync/ && ./sync.sh ) 2>&1',
+      user     => 'root',
+      hour     => $ldap_sync_cron_hour,
+      minute   => $ldap_sync_cron_minute,
+      weekday  => $ldap_sync_cron_weekday,
+      month    => $ldap_sync_cron_month,
+      monthday => $ldap_sync_cron_monthday,
```

Comment:
Suggest: changing parameter to a hash allows condensing these lines to `* => $ldap_sync_cron_parts` (or whatever the parameter name becomes).

Reply:
future enhancement PR